src/sp/it/pl/audio/playback/VlcPlayer.kt

```diff
@@ -164,25 +164,44 @@ class VlcPlayer: GeneralPlayer.Play {
 
     companion object: KLogging() {
 
-        private fun discoverVlc(location: File) = NativeDiscovery(computeVlcDiscoverer(location), DefaultWindowsNativeDiscoveryStrategy(), DefaultLinuxNativeDiscoveryStrategy(), DefaultMacNativeDiscoveryStrategy()).discover()
-
-        private fun computeVlcDiscoverer(location: File) = object: DefaultWindowsNativeDiscoveryStrategy() {
-            override fun onGetDirectoryNames(directoryNames: MutableList<String>) {
-                directoryNames.clear()
-                directoryNames += location.absolutePath
-            }
+        private fun discoverVlc(location: File): Boolean {
+            val loc = location.absolutePath
+            return NativeDiscovery(WindowsDiscoverer(loc), LinuxDiscoverer(loc), MacDiscoverer(loc)).discover()
```

Comment:
Im thinking whether we want to run all discoverers and not only the one for the Os we are running on. What do you think?

If we do, then I'd add a private extension property Os.vlcDiscoverer, which has a when expression which decides the discoverer to use depending on the Os.current.

Reply:
Oh btw, the name is misleading, it does not discoverVlc, only compute the discoverer I think the `computeVlcDiscoverer` is better fit for the `discoverVlc` method.

Reply:
Actually, NativeDiscovery automatically filters the OS and only runs the discoveries fitting the current OS. I looked into the code.

But it is discovering? It instantly calls `discover`.

Reply:
So it is not a problem to add in all discoverers and only the one pertaining to current platform will be used?
If thats the case this is a non issue and you can resolve this.

If it instantly calls discover then it is fine. But that is very weird - how can a constructor already call the discover() method? Makes no sense to me. Are you absolutely sure?

Reply:
1. Yes it does
2. Look at the code: `return NativeDiscovery(...).discover()`

Reply:
ouch, there is discover() at the end. Hm. Thats... sneaky. Damn github's narrow comment section.